add badge component

saadahmadhere opened the pull request:
![image](https://user-images.githubusercontent.com/52596800/152534119-6bd9c70d-ab48-41d3-950d-dd37a7b04bbf.png)
![image](https://user-images.githubusercontent.com/52596800/152534149-688562f9-2e27-4a74-a7a6-188bb5caaf95.png)

https://user-images.githubusercontent.com/52596800/152534720-420d3525-895f-4571-b6b4-eadad586179e.mp4

saadahmadhere committed:
add badge

hinanshisuthar commented on components/badge.css:
```diff
@@ -0,0 +1,71 @@
+.badge{
+    position: absolute;
+    border-radius: 50%;
+    border: 1px solid #fff;
```
can use color variables if youve defined

saadahmadhere replied:
done!

hinanshisuthar reviewed (commented):
Also, try for responsiveness of the code snippet. By setting inline width: 100% to the iframe. Other than that looks good.

aryan-13 commented on components/badge.css:
```diff
@@ -0,0 +1,71 @@
+    justify-content: center;
+    align-items: center;
+    font-size: 10px;
+    color: #fff;
```
This color can also be put in a variable

saadahmadhere replied:
done!

aryan-13 reviewed (changes requested):
Else everything looks good

saadahmadhere committed:
fix: add variable to color in badge

- add color variables for text inside badge.
- fixed the responsiveness of iframes
- add letter spacing in index.css
- update color names in base.css
- update doc.css, sidebar.css
